### PR TITLE
Add Licensing JS Package

### PR DESCRIPTION
**Pull request by cbauerman**
#### Changes proposed in this Pull Request:

* Add new Licensing JS Package to power license activation flow in Plugins

#### Jetpack product discussion

* p1HpG7-cSg-p2

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Package is not connected to any plugin in this PR
    * Follow-up PRs will contain more detailed instructions for testing UX.
* Verify tests pass `cd projects/js-packages/licensing/ && pnpm test`

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,6 +2,7 @@
 /.sass-cache/
 /node_modules
 vendor/
+/.nova/
```
Comment (cbauerman) [inline .gitignore:5]: added additional IDE config ignoring here

File: projects/js-packages/licensing/components/activation-screen-controls/style.scss [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+	&:active {
+		background: #007117;
+		border-color: #007117;
+		color: var( --jp-white );
+	}
+
+	&[disabled],
+	&:disabled {
+		background: #dcdcde;
```
Comment (rcanepa) [inline projects/js-packages/licensing/components/activation-screen-controls/style.scss:104]: I think we replace `#dcdcde` with `var( --jp-gray )`.

See https://github.com/Automattic/jetpack/blob/59940645adf7ee643b1efb71a866a4f93d2c68b2/projects/js-packages/licensing/components/rna-styles.scss#L14

File: projects/js-packages/licensing/components/activation-screen-controls/style.scss [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+		color: #a7aaad;
+		border-color: #dcdcde;
```
Comment (rcanepa) [inline projects/js-packages/licensing/components/activation-screen-controls/style.scss:106]: Same here.

File: projects/js-packages/licensing/components/activation-screen-illustration/index.jsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ * @returns {React.Component} The `ActivationScreenIllustration` component.
+ */
+const ActivationScreenIllustration = props => {
+	const { imageUrl, showSupportLink = false } = props;
+	return (
+		<div className="jp-license-activation-screen-illustration">
+			<img className="jp-license-activation-screen-illustration__img" src={ imageUrl } alt="" />
+			{ showSupportLink && (
+				<p className="jp-license-activation-screen-illustration__support-link">
+					{ createInterpolateElement( __( 'Do you need help? <a>Contact us.</a>', 'jetpack' ), {
```
Comment (rcanepa) [inline projects/js-packages/licensing/components/activation-screen-illustration/index.jsx:30]: Leaving a note here just to remember that we will need to add a link here.

File: projects/js-packages/licensing/components/activation-screen-success-info/style.scss [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+	&:active {
+		background: #007117;
+		border-color: #007117;
+		color: var( --jp-white );
+	}
+
+	&[disabled],
+	&:disabled {
+		background: #dcdcde;
```
Comment (rcanepa) [inline projects/js-packages/licensing/components/activation-screen-success-info/style.scss:59]: Same as above.

**Review by rcanepa — APPROVED** [review]
LGTM.

Code changes look fine and clearly establish the basis for the rest of the pieces of the UI that will come later in follow-up PRs. I have to say also that already tested most of these changes when they were part of a previous PR (closed in favor of this one).

Awesome work, Caleb!

File: projects/js-packages/licensing/components/activation-screen-controls/error-gridicon.jsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+const ErrorGridicon = () => {
```
Comment (jeherve) [inline projects/js-packages/licensing/components/activation-screen-controls/error-gridicon.jsx:6]: Would it be an option to import the icon from `@wordpress/icons` instead?

File: projects/js-packages/licensing/components/activation-screen-controls/index.jsx [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+ */
+const ActivationScreenControls = props => {
+	const { activateLicense, disabled, license, licenseError, onLicenseChange, siteUrl } = props;
+
+	const hasLicenseError = licenseError !== null && licenseError !== undefined;
+
+	return (
+		<div className="jp-license-activation-screen-controls">
+			<div className="jp-license-activation-screen-controls--content">
+				<JetpackLogo showText={ false } height={ 48 } logoColor="#069E08" />
```
Comment (jeherve) [inline projects/js-packages/licensing/components/activation-screen-controls/index.jsx:41]: Do we need to hardcode the color here? `069E08` seems to be the new Jetpack green default logo colour, so I would be tempted to update `JetpackLogo` to use that color by default. I'll do that in a follow-up PR.

```suggestion
				<JetpackLogo showText={ false } height={ 48 } />
```

File: projects/js-packages/licensing/components/activation-screen-controls/style.scss [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+	border-radius: var( --jp-border-radius );
+	color: var( --jp-white );
+
+	padding: 13.5px 45px 13.5px 45px;
+	min-height: 48px;
+
+	margin-top: 24px;
+
+	&:hover {
+		background-color: #007117;
```
Comment (jeherve) [inline projects/js-packages/licensing/components/activation-screen-controls/style.scss:84]: I believe this is `$jetpack_green_60`. Do you think you could add that reference to `rna-styles`?

File: projects/js-packages/licensing/components/activation-screen-controls/style.scss [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+		border-color: #007117;
+		color: var( --jp-white );
+	}
+
+	&:focus {
+		background-color: #005b18;
```
Comment (jeherve) [inline projects/js-packages/licensing/components/activation-screen-controls/style.scss:90]: This is `$jetpack_green_70` I believe, so we could extract it too.

File: projects/js-packages/licensing/components/activation-screen-success-info/product-details/jetpack-backup-daily.jsx [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+ * @returns {React.Component} The `JetpackBackupDailyProductDetails` component.
+ */
+const JetpackBackupDailyProductDetails = props => {
+	const { dashboardUrl } = props;
+	return (
+		<div>
+			<h1>
+				{ __( 'Your Jetpack Daily Backup is active!', 'jetpack' ) }{ ' ' }
+				{ String.fromCodePoint( 0x1f389 ) }
+				{ /* Celebration emoji 🎉 */ }
```
Comment (jeherve) [inline projects/js-packages/licensing/components/activation-screen-success-info/product-details/jetpack-backup-daily.jsx:23]: Do you think we could add the space and the emoji in the translatable string with a `sprintf`, so things are easier to translate?

File: projects/js-packages/licensing/components/activation-screen-success-info/product-details/jetpack-security-daily.jsx [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+ * @returns {React.Component} The `JetpackBackupDailyProductDetails` component.
+ */
+const JetpackSecurityDailyProductDetails = props => {
+	const { dashboardUrl } = props;
+	return (
+		<div>
+			<h1>
+				{ __( 'Your Jetpack Daily Security is active!', 'jetpack' ) }{ ' ' }
+				{ String.fromCodePoint( 0x1f389 ) }
+				{ /* Celebration emoji 🎉 */ }
```
Comment (jeherve) [inline projects/js-packages/licensing/components/activation-screen-success-info/product-details/jetpack-security-daily.jsx:23]: Same as before.

File: projects/js-packages/licensing/index.jsx [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+*/
```
Comment (jeherve) [inline projects/js-packages/licensing/index.jsx:15]: I don't think we need to define the license here; we automatically attach a license when a project is released externally.

**Review by jeherve — COMMENTED** [review]
I gave this a first look. Thanks for adding the tests, that makes things a lot easier! I had a few remarks and questions, see below.

File: projects/js-packages/licensing/components/activation-screen-controls/error-gridicon.jsx (hunk 6 above)
Comment (cbauerman) [reply projects/js-packages/licensing/components/activation-screen-controls/error-gridicon.jsx:6]: `warning` from `@wordpress/icons` ( the bottom icon ) seems like an acceptable substitute to the original ( the top icon )
Screen Shot 2021-11-15 at 10 14 10 AM

**Review by jeherve — APPROVED** [review]
This looks good to me. Let's merge this! 🚢 ⭐

**Comment by anomiex** [comment]
r235224-wpcom

**Pull request merged**